core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -1755,24 +1755,30 @@ private static void splitJoinCondition(
    * can rewritten in expanded form as
    * {@code t1.key = t2.key OR (t1.key IS NULL AND t2.key IS NULL)}.
    *
-   * @param call       Function expression to try collapsing
+   * @param rexNode       Function expression to try collapsing
    * @param rexBuilder {@link RexBuilder} instance to create new {@link RexCall} instances.
    * @return If the given function is an expanded IS NOT DISTINCT FROM function call,
    *         return a IS NOT DISTINCT FROM function call. Otherwise return the input
    *         function call as it is.
    */
-  public static RexCall collapseExpandedIsNotDistinctFromExpr(final RexCall call,
+  public static RexNode collapseExpandedIsNotDistinctFromExpr(final RexNode rexNode,
```

Comment:
We can't change the signature of public methods to avoid breaking backward compatibility, you can create a new method with this signature and call the other method with an appropriate cast, and mark the original method as deprecated (grep "@Deprecated // to be removed before 2.0" and you will find many examples).